### PR TITLE
sanitycheck: fix --failed-only handling

### PR DESCRIPTION
**Pull request by nashif**
- Report build errors as errors, not test failures
- Do not try and build/run tests with build failures
- Fix issue with empty reports when running --only-failed
- Report build errors in the detailed and target reports

### CODE REVIEW

File: scripts/sanity_chk/sanitylib.py [hunk 1]
```diff
@@ -2999,10 +3000,21 @@ def xunit_report(self, filename, platform=None, full_report=False, append=False)
             # When we re-run the tests, we re-use the results and update only with
             # the newly run tests.
             if os.path.exists(filename) and append:
-                eleTestsuite = eleTestsuites.findall(f'testsuite/[@name="{p}"]')[0]
-                eleTestsuite.attrib['failures'] = "%d" % fails
-                eleTestsuite.attrib['errors'] = "%d" % errors
-                eleTestsuite.attrib['skip'] = "%d" % skips
+                ts = eleTestsuites.findall(f'testsuite/[@name="{p}"]')
+                print(ts)
```
Comment (PerMac) [inline scripts/sanity_chk/sanitylib.py:3004]: I guess this print is DBG leftover?

**Review by PerMac — CHANGES_REQUESTED** [review]
Looks good besides this print() in l:3004

**Review by galak — APPROVED** [review]

**Comment by PerMac** [comment]
Are this fails introduced by this commit or the issues with quemu_arc discussed on Slack?

**Review by PerMac — APPROVED** [review]
Passed after rebuild

**Pull request merged**